builder/role_image.go

```diff
@@ -50,25 +50,120 @@ type RoleImageBuilder struct {
 }
 
 // NewRoleImageBuilder creates a new RoleImageBuilder
-func NewRoleImageBuilder(repository, compiledPackagesPath, targetPath, version, fissileVersion string, ui *termui.UI) *RoleImageBuilder {
+func NewRoleImageBuilder(repository, compiledPackagesPath, targetPath, version, fissileVersion string, ui *termui.UI) (*RoleImageBuilder, error) {
+	if err := os.MkdirAll(targetPath, 0755); err != nil {
+		return nil, err
+	}
 	return &RoleImageBuilder{
 		repository:           repository,
 		compiledPackagesPath: compiledPackagesPath,
 		targetPath:           targetPath,
 		version:              version,
 		fissileVersion:       fissileVersion,
 		ui:                   ui,
+	}, nil
+}
+
+// CreatePackagesDockerBuildDir generates a Dockerfile and assets for the shared packages layer and returns a path to the dir
+func (r *RoleImageBuilder) CreatePackagesDockerBuildDir(roleManifest *model.RoleManifest, lightManifestPath, darkManifestPath string) (string, error) {
+	if len(roleManifest.Roles) == 0 {
+		return "", fmt.Errorf("No roles to build")
+	}
+
+	succeeded := false
+	buildDir, err := ioutil.TempDir(r.targetPath, "role-packages")
```

Comment:
I'm planning on having a follow-up PR to put it somewhere in the compilation cache's ancestry (so that we reduce copying).

Using a temp dir everywhere was to guard against things going really wrong and accidentally reusing cruft from a previous run; I'd rather we leave things behind on accident (though we shouldn't) than making bad builds.